run_kraken.py

```diff
@@ -12,6 +12,88 @@
 import pyfiglet
 
 
+# Function to publish kraken status to cerberus
+def publish_kraken_status(config, failed_post_scenarios):
+
+    if config["cerberus"]["cerberus_enabled"]:
+        cerberus_url = config["cerberus"]["cerberus_url"]
+        if not cerberus_url:
+            logging.error("url where Cerberus publishes True/False signal "
+                          "is not provided.")
+            sys.exit(1)
+        cerberus_status = requests.get(cerberus_url).content
+        cerberus_status = True if cerberus_status == b'True' else False
+        if not cerberus_status:
+            logging.error("Received a no-go signal from Cerberus, looks like the"
+                          " cluster is unhealthy. Please check the Cerberus report"
+                          " for more details. Test failed.")
+            if failed_post_scenarios:
+                if config['kraken']['exit_on_failure']:
+                    logging.info("Cerberus status is not healthy and post action scenarios "
+                                 "are still failing, exiting kraken run")
+                    sys.exit(1)
+                else:
+                    logging.info("Cerberus status is not healthy and post action scenarios "
+                                 "are still failing")
+        else:
+            logging.info("Received a go signal from Ceberus, the cluster is "
+                         "healthy. Test passed.")
+
+            if failed_post_scenarios:
+                if config['kraken']['exit_on_failure']:
+                    logging.info("Cerberus status is healthy but post action scenarios "
+                                 "are still failing, exiting kraken run")
+                    sys.exit(1)
+                else:
+                    logging.info("Cerberus status is healthy but post action scenarios "
+                                 "are still failing")
+    else:
+        if config['kraken']['exit_on_failure']:
+            if failed_post_scenarios:
+                logging.info("Post action scenarios are failing, exiting kraken run")
+                sys.exit(1)
+
+
+# Perform the post scenario actions to see if components recovered
+def post_actions(kubeconfig_path, scenario, failed_post_scenarios):
+    for failed_scenario in failed_post_scenarios:
```

Comment:
Just double checking my understanding of the logic.
    - Each time a scenario is run, call post_actions() after running the scenario itself
    - If there are any failed post-actions remaining from previous scenarios, run those post-actions again to see if they have cleared
    - If the current scenario array has a second entry, it is treated as a post-action and should either be yaml (powerfulseal) or custom bash.   If it fails, it goes on the list.

Reply:
Yes that is correct, is there a different way that this could be accomplished?

Reply:
That was for my understanding.  I am fine with the approach and the code looks good to me.   I did not run through all of the status publishing scenarios but did test the post actions.   @chaitanyaenr thoughts on the approach?

Reply:
Sounds good to me. The issue might not be relevant to this PR but think we need to separate and print the pass/failures at the end of the iteration. Looks like the iterations are combined together, output from the current run:
```2020-06-26 13:02:40,479 [INFO] Daemon mode not enabled, will run through 2 iterations
2020-06-26 13:02:41,695 [INFO] Scenario: scenarios/etcd.yml has been successfully injected!
2020-06-26 13:02:41,695 [INFO] Waiting for the specified duration: 0
2020-06-26 13:02:42,909 [INFO] scenarios/post_action_etcd.yaml post action checks passed
2020-06-26 13:02:44,136 [INFO] Scenario: scenarios/openshift-kube-apiserver.yml has been successfully injected!
2020-06-26 13:02:44,137 [INFO] Waiting for the specified duration: 0
2020-06-26 13:02:45,354 [INFO] scenarios/post_action_openshift-kube-apiserver.yml post action checks passed
2020-06-26 13:02:46,572 [INFO] Scenario: scenarios/openshift-apiserver.yml has been successfully injected!
2020-06-26 13:02:46,572 [INFO] Waiting for the specified duration: 0
2020-06-26 13:02:47,787 [INFO] scenarios/post_action_openshift-apiserver.yml post action checks passed
2020-06-26 13:02:49,005 [INFO] Scenario: scenarios/etcd.yml has been successfully injected!
2020-06-26 13:02:49,005 [INFO] Waiting for the specified duration: 0
2020-06-26 13:02:50,215 [INFO] scenarios/post_action_etcd.yaml post action checks passed
2020-06-26 13:02:51,435 [INFO] Scenario: scenarios/openshift-kube-apiserver.yml has been successfully injected!
2020-06-26 13:02:51,435 [INFO] Waiting for the specified duration: 0
2020-06-26 13:02:52,653 [INFO] scenarios/post_action_openshift-kube-apiserver.yml post action checks passed
2020-06-26 13:02:53,861 [INFO] Scenario: scenarios/openshift-apiserver.yml has been successfully injected!
2020-06-26 13:02:53,861 [INFO] Waiting for the specified duration: 0
2020-06-26 13:02:55,080 [INFO] scenarios/post_action_openshift-apiserver.yml post action checks passed
```

Reply:
I added a new line at the end of each iteration and printed out the iteration number at the top of each loop. The output will now look like this (I only ran the etcd scenario).... Thoughts?

```
2020-06-26 13:31:03,551 [INFO] Daemon mode not enabled, will run through 2 iterations
2020-06-26 13:31:03,551 [INFO] Executing scenarios for iteration 0
2020-06-26 13:31:12,288 [INFO] Scenario: scenarios/etcd.yml has been successfully injected!
2020-06-26 13:31:12,289 [INFO] Waiting for the specified duration: 0
2020-06-26 13:31:14,054 [INFO] scenarios/post_action_etcd.yaml post action checks passed
2020-06-26 13:31:14,054 [INFO] 
2020-06-26 13:31:14,055 [INFO] Executing scenarios for iteration 1
2020-06-26 13:31:39,194 [INFO] Scenario: scenarios/etcd.yml has been successfully injected!
2020-06-26 13:31:39,194 [INFO] Waiting for the specified duration: 0
2020-06-26 13:31:40,804 [INFO] scenarios/post_action_etcd.yaml post action checks passed
2020-06-26 13:31:40,805 [INFO] 
```